src/main/webapp/script.js

```diff
@@ -39,27 +39,63 @@ function createMap() {
       {center: {lat: 37.4220, lng: -122.084}, zoom: 16});
 }
 
+/** Deprecated */
 async function getGreetingAsyncAwait(){
     const response = await fetch('/data');
     const quote = await response.text();
     document.getElementById('quote-container').innerText = quote;
 }
 
-function getServerMessages(){
-    fetch('/data').then(response => response.json()).then((messages) => {
-        const messagesListElement= document.getElementById('quote-container');
-        const numToDisplay = 4;
-        var i;
-        for(i = 0; i < Math.min(numToDisplay, messages.length); i++){
-            messagesListElement.appendChild(
-            createListElement(messages[i]));
-        }
-    });
+async function getServerMessages(){
+  const commentLimit = document.getElementById('display-number').value;
+
+  const response = await fetch('/data?num-comments='+ commentLimit);
+  const messages = await response.json();
+  const messagesListElement= document.getElementById('quote-container');
+  messagesListElement.innerHTML = '';
+
+  var i;
+  for(i = 0; i < messages.length ; i++){
+    messagesListElement.appendChild(
+    createListElement(messages[i]));
+  }
+  
+   //Clear List Element and Database
+  deleteListElement(messages, messagesListElement);
 }
 
-function createListElement(text) {
+function createListElement(message) {
   const liElement = document.createElement('li');
-  liElement.innerText = text;
+  liElement.innerText = message;
+
+  /*headElement = document.createElement('h2');
+  const headElement.innerTex; */
   return liElement;
 }
 
+function deleteListElement(messages, messagesListElement) {
```

Comment:
It makes more sense to just make the delete button a plain HTML element which sends an HTTP DELETE (or HTTP POST) to the server. In response, the server redirects the browser to the same page with the updated comments list (containing no comments).

This approach reduces the amount of JS, and makes the button more straightforward as it's just in plain HTML.

Reply:
Sure, but I wanted to have the delete button only appear when the wavlets appeared. It was more of a stylistic choice.